auth/genericoauth/provider.go

```diff
@@ -141,9 +142,12 @@ func (GenericTeamProvider) ProviderConstructor(
 	}
 
 	var oauthVerifier verifier.Verifier
-	if genericOAuth.Scope != "" {
+	if genericOAuth.Scope != "" && genericOAuth.Authority = "" {
 		oauthVerifier = NewScopeVerifier(genericOAuth.Scope)
-	} else {
+	} else if genericOAuth.Scope = "" && genericOAuth.Authority != "" {
+		oauthVerifier = NewAuthorityVerifier(genericOAuth.Authority)
+	}
+	else {
```

Comment:
Not blocking merge on this, but `go fmt` would clean this up (may want to look for editor integration too - a lot of them can do it automatically on save).